src/utils/geojson/generateGeoJson.ts

```diff
@@ -1,119 +1,53 @@
 
 /**
  * Generates random features based on the specified options
  *
  * @param options Options for generating random features
  * @returns Array of generated features
  */
-export const generateGeoJson = (options: CollectionGeneratorOptions) => {
-  const featureTypes = Object.keys(options) as FeatureTypes[];
-  const typeProbabilities = featureTypes.reduce((probabilities, type) => {
-    const prob = options[type]?.probability;
-    if (prob !== undefined) {
-      probabilities[type] = prob;
-    }
-    return probabilities;
-  }, {} as { [key in FeatureTypes]: number });
-
-  const features = [];
-  for (let i = 0; i < options.numFeatures; i++) {
-    const type = weightedRandom(featureTypes, typeProbabilities);
-    const option = options[type];
-    if (!option) continue;
-    const properties = option.generateProperties();
-    const maxDistance = option.maxDistance || DEFAULT_MAX_DISTANCE;
-    const minDistance = option.minDistance || DEFAULT_MIN_DISTANCE;
-    let coordinates;
-    switch (type) {
-      case 'Point':
-        coordinates = [Math.random() * 360 - 180, Math.random() * 180 - 90];
-        break;
-      case 'Polygon':
-      case 'LineString':
-        const numCoordinates = options[type]?.numCoordinates || 3;
-        coordinates = [];
-        let previousLat = 0;
-        let previousLng = 0;
-        for (let j = 0; j < numCoordinates; j++) {
-          let lat = Math.random() * 180 - 90;
-          let lng = Math.random() * 360 - 180;
-          if (j !== 0) {
-            let latDifference = Math.abs(previousLat - lat);
-            let lngDifference = Math.abs(previousLng - lng);
-            while (
-              latDifference > maxDistance ||
-              lngDifference > maxDistance ||
-              latDifference < minDistance ||
-              lngDifference < minDistance
-            ) {
-              lat = Math.random() * 180 - 90;
-              lng = Math.random() * 360 - 180;
-              latDifference = Math.abs(previousLat - lat);
-              lngDifference = Math.abs(previousLng - lng);
-            }
-          }
-          previousLat = lat;
-          previousLng = lng;
-          coordinates.push([lat, lng]);
-        }
-        if (type === 'Polygon') {
-          coordinates = [coordinates];
-        }
-        break;
-    }
-    features.push({ type, properties, coordinates });
+export const generateGeoJson = (options) => {
+  let features;
+  switch (options.type) {
+    case 'Point':
+      const pointGeo = turf.randomPoint(options.numGeometries);
+      features = pointGeo.features.map((geoData) => {
+        Object.assign(geoData.geometry, {
+          properties: options.generateProperties(),
+        });
+        return geoData.geometry;
+      });
+      break;
+    case 'Polygon':
+      const polygons = turf.randomPolygon(options.numGeometries, {
+        max_radial_length: 2,
+      });
+      const simplifiedPolygons = turf.simplify(polygons, {
+        tolerance: 0.9,
```

Comment:
@ketanreliefapps this should be in the API endpoint

we're using mock data at the moment, but in the future, we should be able to simplify geometry of the geojson when querying it ( geojson produced from data of the database )

Simplification should be put in API endpoint so, and it should be possible as an User when interacting with the endpoint to tell what would be the simplification

Something else you can think about:
- is it possible to know the size of polygons ? -> and so, propose a simplification
- what is the possible range of simplification ? what is the maximum, what is the minimum ?
- what is the high quality there ?

Reply:
@AntoineRelief ,
1) is it possible to know the size of polygons ? -> and so, propose a simplification
**ANS:** currently polygons generate with 11 latitude and longitude by [turfjs](https://turfjs.org/docs/#sample) and after simplify it is remaining 5 to 7

2) what is the possible range of simplification ? what is the maximum, what is the minimum ?
**ANS:** Not because turfjs not provided this type of setting, it is providing only tolerance setting and based on geo json simplified. here documents of simplify geo json https://turfjs.org/docs/#simplify

3) what is the high quality there ?
**ANS:** high quality is whether or not to spend more time to create a higher-quality simplification with a different algorithm
here some screen shot, so you can get idea about quality setting

this is origin geo polygon
![image](https://user-images.githubusercontent.com/111883188/224918163-b8c73c61-683b-4414-b8ee-edd4d6fe3e69.png)

this is simplify geo polygon with high quality setting
![image](https://user-images.githubusercontent.com/111883188/224918339-cad410c0-2045-42b1-ac59-140b728939ae.png)

this is simplify geo polygon without high quality setting
![image](https://user-images.githubusercontent.com/111883188/224918489-fd5317d1-51ff-475c-8748-fabf21dd9720.png)

Reply:
https://github.com/ReliefApplications/oort-backend/pull/553

Reply:
@ketanreliefapps okay so high quality should remain to true I believe

Reply:
@ketanreliefapps another question for this one
can turf give an estimation of the size of a geojson, not in term of polygons but bytes ?